gtk4/src/tree_iter.rs

```diff
@@ -0,0 +1,152 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use cairo::glib::Object;
+use glib::ffi::gpointer;
+use glib::object::GObject;
+use glib::translate::*;
+use glib::ObjectType;
+
+use std::mem;
+use std::os::raw::c_int;
+
+// 0xffffffff is an invalid pointer that is distinct from the null pointer
+// This makes it unlikely this is in here by accident
+const MAGIC_NUMBER_1: usize = 0xffffffff;
+// "rust" in ASCII
+const MAGIC_NUMBER_2: usize = 0x72757374;
+
+#[repr(C)]
+#[derive(Clone)]
+#[doc(alias = "GtkTreeIter")]
+pub struct TreeIter {
```

Comment:
Why not

```rust
#[repr(transparent)]
pub struct TreeIter(ffi::GtkTreeIter);
```
?

Reply:
RGBA in glib does it the same way but sure I can do it that way, nothing wrong with it.

Reply:
Because there the fields are public and that would be ugly to use otherwise